uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/ContactWorkPattern.php

```diff
@@ -188,4 +190,38 @@ public static function getWorkDayType($contactId, DateTime $date) {
     return $workDayTypeId;
   }
 
+  /**
+   * Returns a list of ContactWorkPattern instances for the given $contactID,
+   * which overlap the period enclosed by the given $start and $end dates
+   *
+   * @param int $contactID
+   * @param \DateTime $start
+   * @param \DateTime $end
+   *
+   * @return \CRM_HRLeaveAndAbsences_BAO_ContactWorkPattern[]
```

Comment:
Its also possible for this method to return an empty array if there are no work patterns for the period.
Why not` \CRM_HRLeaveAndAbsences_BAO_ContactWorkPattern[]|[]` ?